tests/test_initializer.py

```diff
@@ -4,7 +4,8 @@
 import numpy
 import pytest
 
-from spox._future import initializer
+from spox import build, initializer
+from spox.opset.ai.onnx import v17 as op
 
```

Comment:
The [ONNX specification](https://github.com/onnx/onnx/blob/main/docs/IR.md) says:

> A list of named tensor values. When an initializer has the same name as a graph input, it specifies a default value for that input. When an initializer has a name different from all graph inputs, it specifies a constant value. The order of the list is unspecified.

How exactly are the names determined for initializers currently? Is it possible that a graph input has the same name as an initializer produced by Spox through this function?

If so, we should add some explicit tests for these edge cases _or_ mangle names in a way that using `initializer` doesn't ever produce the same name as a graph input.
1. Input named "X" and an initializer also named "X", with compatible shape + type. What will likely happen is that the value of the input is used but this is a discrepancy with using `constant` which users might not be aware of. We should at a minimum communicate this difference in the docstring for `initializer` or simply make this impossible when using the public function from Spox.
2. Input named "X" and initializer also named "X", with different types (does this cause the model checker to complain)

Scoping seems weird with initializers (unlike constants). We should test these edge cases explicitly.
> In models with IR version >= 4, in nested subgraphs used as attribute values, users MUST NOT use the same name as both a subgraph initializer and subgraph input unless the corresponding op's specification explicitly allows it.

and
> In models with IR version <= 3, users MAY use the same name as both a subgraph initializer and subgraph input, but this is restricted to support constants via initializers that are not intended to correspond to any actual inputs passed from the node into the subgraph.

Do we need to do some name mangling for subgraphs (cf. inlining, can probably share code) here?

Reply:
It seems most suitable at this stage to prevent initializer and argument from sharing identical names.

I've tackled the other issues by incorporating more tests. Please don't hesitate to extend the discussion if there are any other edge cases that we've overlooked.